pkg/cloudprovider/azure.go

```diff
@@ -577,17 +577,23 @@ func (a *Azure) getAddressPrefixes(networkInterface armnetwork.Interface) ([]*st
 	// FIXME: This might not work for IPv6.
 	if virtualNetwork.Properties != nil && virtualNetwork.Properties.Subnets != nil {
 		for _, vns := range virtualNetwork.Properties.Subnets {
-			if vns.Name != nil && vns.Properties.AddressPrefix != nil &&
-				ptr.Deref(vns.Name, "") == subnetName {
-				return []*string{vns.Properties.AddressPrefix}, nil
+			if vns.Name != nil && ptr.Deref(vns.Name, "") == subnetName {
```

Comment:
so we are trying to network interface addresses+prefixes and if AddressPrefix is set, we chose that. 

What happens if its dualstack cluster and therefore the network interface may have v4 and v6? 
we may only get address + prefix for a single stack. 

Why not favour `AddressPrefixes` over `AddressPrefix` or get the union of the two if the API isnt clearly defined?

Reply:
how did the reporter end up with `AddressPrefixes` set and not `AddressPrefix` - i assume we have CI for basic sanity testing with a config we support so i am curious how the reporter got the subnet with `AddressPrefixes` set.